bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/utils/ClangdUtils.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   See git history
+ *******************************************************************************/
+
+package org.eclipse.cdt.lsp.clangd.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Platform;
+
+public class ClangdUtils {
+	public static final String format_file = ".clang-format"; //$NON-NLS-1$
+
+	/**
+	 * Checks if a .clang-format file exists in the workspace root directory. Creates it if not.
+	 * This ensures that the user has a default formatting similar to the CDT default K&R formatter
+	 *
+	 */
+	public void checkWorkspaceClangFormatFile(IWorkspace workspace) {
+		var path = workspace.getRoot().getLocation().append(format_file).toOSString();
```

Comment:
The location of file is something to design more carefully. Usually plugins do not  store its data to the root of workspace, but use plugin-specific areas.

Reply:
The reason why we store the `.clang-format` file in the workspace root is: clangd searches parent directories for `.clang-format` files. Putting a  `.clang-format` file in the root directory ensures that workspace wide preference will be used for formatting via clangd. I think it's a smart way (according to the convention over configuration paradigm) to tell clangd which `.clang-format` file should be used.

Reply:
I do respect convention over configuration paradigm, but it is very doubtful to expect that workspace root will be the parent for project from file system perspective. However, if you know for sure that this is how users work with CDT LSP, I'm unlikely to argue.

Reply:
> it is very doubtful to expect that workspace root will be the parent for project from file system perspective

Ah, good point. I have expected that.

Reply:
I assume that in the majority the projects parent directory is equal to  the workspace root directory, but it cannot be guaranteed. Therefor my proposal would be to create the `.clang-format` file only in the projects root directory.